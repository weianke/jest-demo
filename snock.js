export const gennerateConfig = () => {
  return {
    serve: 'http://localhost',
    port: 8080,
    domain: 'localhost',
    time:  new Date(),
    nihap: '223222'
  }
}

export const gennerateAnotherConfig = () => {
  return {
    serve: 'http://localhost',
    port: 8080,
    domain: 'localhost',
    token: '32kr',
    nihap: '22wwew3'
  }
}