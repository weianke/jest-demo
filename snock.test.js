import { gennerateConfig, gennerateAnotherConfig } from "./snock";

// test('测试 gennerateConfig 函数', () => {
// expect(gennerateConfig()).toMatchSnapshot({
//   time: expect.any(Date)
// });
// })

// test('测试 gennerateAnotherConfig 函数', () => {
//   expect(gennerateAnotherConfig()).toMatchSnapshot();
// })

test("测试 gennerateConfig 函数", () => {
  expect(gennerateConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date)
    },
    `
    Object {
      "domain": "localhost",
      "nihap": "223222",
      "port": 8080,
      "serve": "http://localhost",
      "time": Any<Date>,
    }
  `
  );
});

test("测试 gennerateAnotherConfig 函数", () => {
  expect(gennerateAnotherConfig()).toMatchInlineSnapshot(`
    Object {
      "domain": "localhost",
      "nihap": "22wwew3",
      "port": 8080,
      "serve": "http://localhost",
      "token": "32kr",
    }
  `);
});
