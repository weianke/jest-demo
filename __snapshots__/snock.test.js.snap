// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 gennerateAnotherConfig 函数 1`] = `
Object {
  "domain": "localhost",
  "nihap": "22wwew3",
  "port": 8080,
  "serve": "http://localhost",
  "token": "32kr",
}
`;

exports[`测试 gennerateConfig 函数 1`] = `
Object {
  "domain": "localhost",
  "nihap": "223",
  "port": 8080,
  "serve": "http://localhost",
  "time": Any<Date>,
}
`;
